sunvox: Refactor, add myself as maintainer

OPNA2608 opened the pull request:
## Description of changes

- Use `autoPatchelfHook` instead of manually patching Linux binaries
- Support more platforms (ARM Linux, Darwin)
- Patch more binaries (x86_64 & aarch64 has `sunvox_opengl` variant, other Linux platforms have others / even more)
- Support JACK
- Add myself as maintainer

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/3034

SuperSandro2000 commented on pkgs/applications/audio/sunvox/default.nix:
```diff
@@ -1,49 +1,88 @@
     description = "Small, fast and powerful modular synthesizer with pattern-based sequencer";
     license = licenses.unfreeRedistributable;
     sourceProvenance = with sourceTypes; [ binaryNativeCode ];
     homepage = "http://www.warmplace.ru/soft/sunvox/";
```
```suggestion
    homepage = "https://www.warmplace.ru/soft/sunvox/";
```

SuperSandro2000 commented on pkgs/applications/audio/sunvox/default.nix:
```diff
@@ -1,49 +1,88 @@
+    mv * $out/share/sunvox/
+
+  '' + lib.optionalString stdenv.hostPlatform.isLinux ''
+    for binary in $(find $out/share/sunvox/sunvox/${bindir}/ -type f -executable); do
+      ln -s $binary $out/bin/$(basename $binary)
+    done
```
Shouldn't we move them?

OPNA2608 replied:
I don't see why we *must*, especially since we can't do it on Darwin, and properly cleaning everything up post-move is more cumbersome, but if you insist.

SuperSandro2000 commented on pkgs/applications/audio/sunvox/default.nix:
```diff
@@ -1,49 +1,88 @@
+  ];
 
-  unpackPhase = "unzip $src";
+  runtimeDependencies = lib.optionals stdenv.hostPlatform.isLinux [
```
That is used by autoPatchelfHook, right?

OPNA2608 replied:
[Yes.](https://nixos.org/manual/nixpkgs/stable/#setup-hook-autopatchelfhook)

OPNA2608 replied:
I don't understand the question? It *will* be in the rpath once put into `runtimeDependencies`.

OPNA2608 committed:
sunvox: Refactor

- Use autoPatchelfHook instead of manually patching Linux binaries
- Support more platforms (ARM Linux, Darwin)
- Patch more binaries (x86_64 & aarch64 has sunvox_opengl variant, other Linux platforms have others / even more)
- Support JACK
- Add myself as maintainer

OPNA2608 committed:
sunvox: Move to pkgs/by-name

pbsds reviewed (approved):
Result of `nixpkgs-review pr 246544` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>sunvox</li>
  </ul>
</details>
